src/sentry/models/group.py

```diff
@@ -45,6 +54,28 @@ class GroupStatus(object):
 class GroupManager(BaseManager):
     use_for_related_fields = True
 
+    def by_qualified_short_id(self, org, short_id):
+        match = _short_id_re.match(short_id.strip())
+        if match is None:
+            return
```

Comment:
Not sure what you mean by "should not validate here". Surely the entire point of this method that it can be used safely from all places based on arbitrary user input.  All query methods that look up something are supposed to work like that.  As far as the caller is concerned an invalid short ID is not invalid but purely does not exist.